common/src/verify.rs

```diff
@@ -149,23 +149,17 @@ impl CommitSequenceVerifier {
                 },
             ) => {
                 verify_header_to_header(&self.header, block_header)?;
-                // Verify block body
-                if *transaction_merkle_root != block_header.tx_merkle_root {
-                    return Err(Error::InvalidArgument(format!(
-                        "invalid transaction merkle root hash: expected {}, got {}",
-                        transaction_merkle_root, block_header.tx_merkle_root
-                    )));
-                }
-                // Verify commit hash
-                if self.commit_hash != block_header.commit_hash {
+                // Verify commit merkle root
+                let commit_merkle_root = BlockHeader::calculate_commit_merkle_root(&self.commits);
+                if commit_merkle_root != block_header.commit_merkle_root {
                     return Err(Error::InvalidArgument(format!(
-                        "invalid block commit hash: expected {}, got {}",
-                        self.commit_hash, block_header.commit_hash
+                        "invalid commit merkle root: expected {}, got {}",
+                        commit_merkle_root, block_header.commit_merkle_root
                     )));
-                }
+                };
                 self.header = block_header.clone();
                 self.phase = Phase::Block;
-                self.commit_hash = Hash256::hash(format!("{}", block_header.height + 1));
+                self.commits = vec![];
```

Comment:
We erased information about block_header's height. Where do we store height information now?

Reply:
The information of block_header's height is still there inside `self.header`.
It was just a base hash(?) for calculating commit hash of the block.